{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "def PRSice2(gwas_path:str,\n",
    "            disease:str,\n",
    "            clump_r2=0.1,\n",
    "            ckb='',\n",
    "            output_prefix=''):\n",
    "\n",
    "    os.system(f\"Rscript PRSice.R \\\n",
    "                    --prsice PRSice_linux \\\n",
    "                    --base {gwas_path} \\\n",
    "                    --target ../../target_data/target.data.impute \\\n",
    "                    --binary-target T \\\n",
    "                    --clump-r2 {clump_r2} \\\n",
    "                    --pheno ../../results/{disease}/samples.txt \\\n",
    "                    --keep ../../results/{disease}/samples.txt \\\n",
    "                    --cov ../../target_data/covar_list.tsv \\\n",
    "                    --cov-col @PC[1-10],Sex,YoB \\\n",
    "                    --beta \\\n",
    "                    --print-snp \\\n",
    "                    --score sum \\\n",
    "                    --seed 68 \\\n",
    "                    --out ../../results/{disease}/PRSice2{ckb}.{disease}.clumpr2.{clump_r2}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def PRSice2_new(gwas_path:str,\n",
    "            disease:str,\n",
    "            clump_r2=0.1,\n",
    "            output_prefix='',\n",
    "            covar_cols='@PC[1-10],Sex,Age'):\n",
    "\n",
    "    os.system(f\"Rscript PRSice.R \\\n",
    "                    --prsice PRSice_linux \\\n",
    "                    --base {gwas_path} \\\n",
    "                    --target ../../target_data/target.data.impute\\\n",
    "                    --binary-target T \\\n",
    "                    --clump-r2 {clump_r2} \\\n",
    "                    --pheno ../../results/{disease}/samples.txt \\\n",
    "                    --keep ../../results/{disease}/samples.txt \\\n",
    "                    --cov ../../target_data/covar_list.tsv \\\n",
    "                    --cov-col {covar_cols} \\\n",
    "                    --beta \\\n",
    "                    --print-snp \\\n",
    "                    --score sum \\\n",
    "                    --seed 68 \\\n",
    "                    --out ../../results/{disease}/{output_prefix}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "clump_r2_list = [0.1, 0.4, 0.7, 0.9]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. Breast cancer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 16:53:04\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/breast_cancer/hum0197.v3.BBJ.BC.v1/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex,Age \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/breast_cancer/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/breast_cancer/PRSice2.BC.clumpr2.0.1.age \\\n",
      "    --pheno ../../results/breast_cancer/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/breast_cancer/hum0197.v3.BBJ.BC.v1/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "10347822 variant(s) observed in base file, with: \n",
      "1489232 ambiguous variant(s) excluded \n",
      "8858590 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "610 founder(s) included \n",
      "\n",
      "1433038 variant(s) not found in previous data \n",
      "4172028 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/breast_cancer/samples.txt \n",
      "Column Name of Sample ID: 674335405400+674335405400 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 129906 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "511 control(s) \n",
      "99 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "Age\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 610 sample(s) included in \n",
      "the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 16:54:29\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/breast_cancer/hum0197.v3.BBJ.BC.v1/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/breast_cancer/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/breast_cancer/PRSice2.BC.clumpr2.0.1.no.age \\\n",
      "    --pheno ../../results/breast_cancer/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/breast_cancer/hum0197.v3.BBJ.BC.v1/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "10347822 variant(s) observed in base file, with: \n",
      "1489232 ambiguous variant(s) excluded \n",
      "8858590 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "610 founder(s) included \n",
      "\n",
      "1433038 variant(s) not found in previous data \n",
      "4172028 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/breast_cancer/samples.txt \n",
      "Column Name of Sample ID: 674335405400+674335405400 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 129906 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "511 control(s) \n",
      "99 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 610 sample(s) included in \n",
      "the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 16:55:55\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/breast_cancer/hum0197.v3.BBJ.BC.v1/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.400000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex,Age \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/breast_cancer/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/breast_cancer/PRSice2.BC.clumpr2.0.4.age \\\n",
      "    --pheno ../../results/breast_cancer/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/breast_cancer/hum0197.v3.BBJ.BC.v1/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "10347822 variant(s) observed in base file, with: \n",
      "1489232 ambiguous variant(s) excluded \n",
      "8858590 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "610 founder(s) included \n",
      "\n",
      "1433038 variant(s) not found in previous data \n",
      "4172028 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/breast_cancer/samples.txt \n",
      "Column Name of Sample ID: 674335405400+674335405400 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 292187 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "511 control(s) \n",
      "99 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "Age\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 610 sample(s) included in \n",
      "the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 16:57:43\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/breast_cancer/hum0197.v3.BBJ.BC.v1/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.400000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/breast_cancer/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/breast_cancer/PRSice2.BC.clumpr2.0.4.no.age \\\n",
      "    --pheno ../../results/breast_cancer/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/breast_cancer/hum0197.v3.BBJ.BC.v1/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "10347822 variant(s) observed in base file, with: \n",
      "1489232 ambiguous variant(s) excluded \n",
      "8858590 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "610 founder(s) included \n",
      "\n",
      "1433038 variant(s) not found in previous data \n",
      "4172028 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/breast_cancer/samples.txt \n",
      "Column Name of Sample ID: 674335405400+674335405400 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 292187 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "511 control(s) \n",
      "99 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 610 sample(s) included in \n",
      "the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 16:59:33\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/breast_cancer/hum0197.v3.BBJ.BC.v1/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.700000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex,Age \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/breast_cancer/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/breast_cancer/PRSice2.BC.clumpr2.0.7.age \\\n",
      "    --pheno ../../results/breast_cancer/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/breast_cancer/hum0197.v3.BBJ.BC.v1/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "10347822 variant(s) observed in base file, with: \n",
      "1489232 ambiguous variant(s) excluded \n",
      "8858590 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "610 founder(s) included \n",
      "\n",
      "1433038 variant(s) not found in previous data \n",
      "4172028 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/breast_cancer/samples.txt \n",
      "Column Name of Sample ID: 674335405400+674335405400 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 527424 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "511 control(s) \n",
      "99 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "Age\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 610 sample(s) included in \n",
      "the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 17:01:55\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/breast_cancer/hum0197.v3.BBJ.BC.v1/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.700000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/breast_cancer/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/breast_cancer/PRSice2.BC.clumpr2.0.7.no.age \\\n",
      "    --pheno ../../results/breast_cancer/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/breast_cancer/hum0197.v3.BBJ.BC.v1/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "10347822 variant(s) observed in base file, with: \n",
      "1489232 ambiguous variant(s) excluded \n",
      "8858590 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "610 founder(s) included \n",
      "\n",
      "1433038 variant(s) not found in previous data \n",
      "4172028 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/breast_cancer/samples.txt \n",
      "Column Name of Sample ID: 674335405400+674335405400 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 527424 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "511 control(s) \n",
      "99 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 610 sample(s) included in \n",
      "the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 17:04:20\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/breast_cancer/hum0197.v3.BBJ.BC.v1/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.900000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex,Age \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/breast_cancer/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/breast_cancer/PRSice2.BC.clumpr2.0.9.age \\\n",
      "    --pheno ../../results/breast_cancer/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/breast_cancer/hum0197.v3.BBJ.BC.v1/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "10347822 variant(s) observed in base file, with: \n",
      "1489232 ambiguous variant(s) excluded \n",
      "8858590 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "610 founder(s) included \n",
      "\n",
      "1433038 variant(s) not found in previous data \n",
      "4172028 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/breast_cancer/samples.txt \n",
      "Column Name of Sample ID: 674335405400+674335405400 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 881644 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "511 control(s) \n",
      "99 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "Age\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 610 sample(s) included in \n",
      "the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 17:07:46\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/breast_cancer/hum0197.v3.BBJ.BC.v1/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.900000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/breast_cancer/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/breast_cancer/PRSice2.BC.clumpr2.0.9.no.age \\\n",
      "    --pheno ../../results/breast_cancer/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/breast_cancer/hum0197.v3.BBJ.BC.v1/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "10347822 variant(s) observed in base file, with: \n",
      "1489232 ambiguous variant(s) excluded \n",
      "8858590 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "610 founder(s) included \n",
      "\n",
      "1433038 variant(s) not found in previous data \n",
      "4172028 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/breast_cancer/samples.txt \n",
      "Column Name of Sample ID: 674335405400+674335405400 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 881644 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "511 control(s) \n",
      "99 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 610 sample(s) included in \n",
      "the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    }
   ],
   "source": [
    "# for gwas EAS (BBJ+)\n",
    "covar_name = ['age', 'no.age']\n",
    "for c in clump_r2_list:\n",
    "        for i, covar_cols in enumerate(['@PC[1-10],Sex,Age', '@PC[1-10],Sex']):\n",
    "                prefix = f'PRSice2.BC.clumpr2.{c}.{covar_name[i]}'\n",
    "                PRSice2_new(gwas_path='../../sum_stats_data/breast_cancer/hum0197.v3.BBJ.BC.v1/gwas.PRSice.txt',\n",
    "                                disease='breast_cancer', clump_r2=c, output_prefix=prefix, covar_cols=covar_cols)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2. Colorectal cancer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 17:11:06\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/colorectal_cancer/hum0197.v3.BBJ.CC.v1/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex,Age \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/colorectal_cancer/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/colorectal_cancer/PRSice2.CRC.clumpr2.0.1.age \\\n",
      "    --pheno ../../results/colorectal_cancer/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/colorectal_cancer/hum0197.v3.BBJ.CC.v1/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "10346101 variant(s) observed in base file, with: \n",
      "1489522 ambiguous variant(s) excluded \n",
      "8856579 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1108 founder(s) included \n",
      "\n",
      "1443745 variant(s) not found in previous data \n",
      "4161321 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/colorectal_cancer/samples.txt \n",
      "Column Name of Sample ID: 018843557900+018843557900 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 129926 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "100 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "Age\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1108 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 17:12:36\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/colorectal_cancer/hum0197.v3.BBJ.CC.v1/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/colorectal_cancer/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/colorectal_cancer/PRSice2.CRC.clumpr2.0.1.no.age \\\n",
      "    --pheno ../../results/colorectal_cancer/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/colorectal_cancer/hum0197.v3.BBJ.CC.v1/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "10346101 variant(s) observed in base file, with: \n",
      "1489522 ambiguous variant(s) excluded \n",
      "8856579 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1108 founder(s) included \n",
      "\n",
      "1443745 variant(s) not found in previous data \n",
      "4161321 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/colorectal_cancer/samples.txt \n",
      "Column Name of Sample ID: 018843557900+018843557900 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 129926 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "100 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1108 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 17:14:01\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/colorectal_cancer/hum0197.v3.BBJ.CC.v1/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.400000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex,Age \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/colorectal_cancer/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/colorectal_cancer/PRSice2.CRC.clumpr2.0.4.age \\\n",
      "    --pheno ../../results/colorectal_cancer/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/colorectal_cancer/hum0197.v3.BBJ.CC.v1/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "10346101 variant(s) observed in base file, with: \n",
      "1489522 ambiguous variant(s) excluded \n",
      "8856579 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1108 founder(s) included \n",
      "\n",
      "1443745 variant(s) not found in previous data \n",
      "4161321 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/colorectal_cancer/samples.txt \n",
      "Column Name of Sample ID: 018843557900+018843557900 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 291584 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "100 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "Age\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1108 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 17:15:55\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/colorectal_cancer/hum0197.v3.BBJ.CC.v1/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.400000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/colorectal_cancer/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/colorectal_cancer/PRSice2.CRC.clumpr2.0.4.no.age \\\n",
      "    --pheno ../../results/colorectal_cancer/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/colorectal_cancer/hum0197.v3.BBJ.CC.v1/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "10346101 variant(s) observed in base file, with: \n",
      "1489522 ambiguous variant(s) excluded \n",
      "8856579 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1108 founder(s) included \n",
      "\n",
      "1443745 variant(s) not found in previous data \n",
      "4161321 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/colorectal_cancer/samples.txt \n",
      "Column Name of Sample ID: 018843557900+018843557900 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 291584 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "100 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1108 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 17:17:47\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/colorectal_cancer/hum0197.v3.BBJ.CC.v1/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.700000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex,Age \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/colorectal_cancer/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/colorectal_cancer/PRSice2.CRC.clumpr2.0.7.age \\\n",
      "    --pheno ../../results/colorectal_cancer/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/colorectal_cancer/hum0197.v3.BBJ.CC.v1/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "10346101 variant(s) observed in base file, with: \n",
      "1489522 ambiguous variant(s) excluded \n",
      "8856579 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1108 founder(s) included \n",
      "\n",
      "1443745 variant(s) not found in previous data \n",
      "4161321 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/colorectal_cancer/samples.txt \n",
      "Column Name of Sample ID: 018843557900+018843557900 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 526192 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "100 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "Age\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1108 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 17:20:25\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/colorectal_cancer/hum0197.v3.BBJ.CC.v1/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.700000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/colorectal_cancer/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/colorectal_cancer/PRSice2.CRC.clumpr2.0.7.no.age \\\n",
      "    --pheno ../../results/colorectal_cancer/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/colorectal_cancer/hum0197.v3.BBJ.CC.v1/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "10346101 variant(s) observed in base file, with: \n",
      "1489522 ambiguous variant(s) excluded \n",
      "8856579 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1108 founder(s) included \n",
      "\n",
      "1443745 variant(s) not found in previous data \n",
      "4161321 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/colorectal_cancer/samples.txt \n",
      "Column Name of Sample ID: 018843557900+018843557900 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 526192 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "100 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1108 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 17:23:00\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/colorectal_cancer/hum0197.v3.BBJ.CC.v1/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.900000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex,Age \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/colorectal_cancer/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/colorectal_cancer/PRSice2.CRC.clumpr2.0.9.age \\\n",
      "    --pheno ../../results/colorectal_cancer/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/colorectal_cancer/hum0197.v3.BBJ.CC.v1/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "10346101 variant(s) observed in base file, with: \n",
      "1489522 ambiguous variant(s) excluded \n",
      "8856579 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1108 founder(s) included \n",
      "\n",
      "1443745 variant(s) not found in previous data \n",
      "4161321 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/colorectal_cancer/samples.txt \n",
      "Column Name of Sample ID: 018843557900+018843557900 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 878455 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "100 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "Age\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1108 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 17:26:51\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/colorectal_cancer/hum0197.v3.BBJ.CC.v1/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.900000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/colorectal_cancer/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/colorectal_cancer/PRSice2.CRC.clumpr2.0.9.no.age \\\n",
      "    --pheno ../../results/colorectal_cancer/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/colorectal_cancer/hum0197.v3.BBJ.CC.v1/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "10346101 variant(s) observed in base file, with: \n",
      "1489522 ambiguous variant(s) excluded \n",
      "8856579 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1108 founder(s) included \n",
      "\n",
      "1443745 variant(s) not found in previous data \n",
      "4161321 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/colorectal_cancer/samples.txt \n",
      "Column Name of Sample ID: 018843557900+018843557900 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 878455 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "100 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1108 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    }
   ],
   "source": [
    "# BBJ+\n",
    "covar_name = ['age', 'no.age']\n",
    "for c in clump_r2_list:\n",
    "        for i, covar_cols in enumerate(['@PC[1-10],Sex,Age', '@PC[1-10],Sex']):\n",
    "                prefix = f'PRSice2.CRC.clumpr2.{c}.{covar_name[i]}'\n",
    "                PRSice2_new(gwas_path='../../sum_stats_data/colorectal_cancer/hum0197.v3.BBJ.CC.v1/gwas.PRSice.txt',\n",
    "                                disease='colorectal_cancer', clump_r2=c, output_prefix=prefix, covar_cols=covar_cols)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3. Gastric cancer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 17:30:34\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/gastric_cancer/hum0197.v3.BBJ.GC.v1/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex,Age \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/gastric_cancer/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/gastric_cancer/PRSice2.GC.clumpr2.0.1.age \\\n",
      "    --pheno ../../results/gastric_cancer/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/gastric_cancer/hum0197.v3.BBJ.GC.v1/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "10346106 variant(s) observed in base file, with: \n",
      "1489521 ambiguous variant(s) excluded \n",
      "8856585 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "1443718 variant(s) not found in previous data \n",
      "4161348 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/gastric_cancer/samples.txt \n",
      "Column Name of Sample ID: 704107872400+704107872400 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 130053 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "Age\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 17:32:07\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/gastric_cancer/hum0197.v3.BBJ.GC.v1/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/gastric_cancer/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/gastric_cancer/PRSice2.GC.clumpr2.0.1.no.age \\\n",
      "    --pheno ../../results/gastric_cancer/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/gastric_cancer/hum0197.v3.BBJ.GC.v1/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "10346106 variant(s) observed in base file, with: \n",
      "1489521 ambiguous variant(s) excluded \n",
      "8856585 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "1443718 variant(s) not found in previous data \n",
      "4161348 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/gastric_cancer/samples.txt \n",
      "Column Name of Sample ID: 704107872400+704107872400 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 130053 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 17:33:36\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/gastric_cancer/hum0197.v3.BBJ.GC.v1/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.400000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex,Age \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/gastric_cancer/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/gastric_cancer/PRSice2.GC.clumpr2.0.4.age \\\n",
      "    --pheno ../../results/gastric_cancer/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/gastric_cancer/hum0197.v3.BBJ.GC.v1/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "10346106 variant(s) observed in base file, with: \n",
      "1489521 ambiguous variant(s) excluded \n",
      "8856585 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "1443718 variant(s) not found in previous data \n",
      "4161348 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/gastric_cancer/samples.txt \n",
      "Column Name of Sample ID: 704107872400+704107872400 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 292089 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "Age\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 17:35:35\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/gastric_cancer/hum0197.v3.BBJ.GC.v1/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.400000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/gastric_cancer/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/gastric_cancer/PRSice2.GC.clumpr2.0.4.no.age \\\n",
      "    --pheno ../../results/gastric_cancer/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/gastric_cancer/hum0197.v3.BBJ.GC.v1/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "10346106 variant(s) observed in base file, with: \n",
      "1489521 ambiguous variant(s) excluded \n",
      "8856585 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "1443718 variant(s) not found in previous data \n",
      "4161348 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/gastric_cancer/samples.txt \n",
      "Column Name of Sample ID: 704107872400+704107872400 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 292089 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 17:37:31\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/gastric_cancer/hum0197.v3.BBJ.GC.v1/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.700000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex,Age \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/gastric_cancer/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/gastric_cancer/PRSice2.GC.clumpr2.0.7.age \\\n",
      "    --pheno ../../results/gastric_cancer/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/gastric_cancer/hum0197.v3.BBJ.GC.v1/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "10346106 variant(s) observed in base file, with: \n",
      "1489521 ambiguous variant(s) excluded \n",
      "8856585 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "1443718 variant(s) not found in previous data \n",
      "4161348 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/gastric_cancer/samples.txt \n",
      "Column Name of Sample ID: 704107872400+704107872400 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 526655 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "Age\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 17:40:11\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/gastric_cancer/hum0197.v3.BBJ.GC.v1/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.700000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/gastric_cancer/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/gastric_cancer/PRSice2.GC.clumpr2.0.7.no.age \\\n",
      "    --pheno ../../results/gastric_cancer/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/gastric_cancer/hum0197.v3.BBJ.GC.v1/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "10346106 variant(s) observed in base file, with: \n",
      "1489521 ambiguous variant(s) excluded \n",
      "8856585 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "1443718 variant(s) not found in previous data \n",
      "4161348 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/gastric_cancer/samples.txt \n",
      "Column Name of Sample ID: 704107872400+704107872400 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 526655 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 17:42:47\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/gastric_cancer/hum0197.v3.BBJ.GC.v1/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.900000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex,Age \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/gastric_cancer/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/gastric_cancer/PRSice2.GC.clumpr2.0.9.age \\\n",
      "    --pheno ../../results/gastric_cancer/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/gastric_cancer/hum0197.v3.BBJ.GC.v1/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "10346106 variant(s) observed in base file, with: \n",
      "1489521 ambiguous variant(s) excluded \n",
      "8856585 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "1443718 variant(s) not found in previous data \n",
      "4161348 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/gastric_cancer/samples.txt \n",
      "Column Name of Sample ID: 704107872400+704107872400 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 878623 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "Age\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 17:46:39\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/gastric_cancer/hum0197.v3.BBJ.GC.v1/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.900000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/gastric_cancer/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/gastric_cancer/PRSice2.GC.clumpr2.0.9.no.age \\\n",
      "    --pheno ../../results/gastric_cancer/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/gastric_cancer/hum0197.v3.BBJ.GC.v1/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "10346106 variant(s) observed in base file, with: \n",
      "1489521 ambiguous variant(s) excluded \n",
      "8856585 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "1443718 variant(s) not found in previous data \n",
      "4161348 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/gastric_cancer/samples.txt \n",
      "Column Name of Sample ID: 704107872400+704107872400 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 878623 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    }
   ],
   "source": [
    "# BBJ+\n",
    "covar_name = ['age', 'no.age']\n",
    "for c in clump_r2_list:\n",
    "        for i, covar_cols in enumerate(['@PC[1-10],Sex,Age', '@PC[1-10],Sex']):\n",
    "                prefix = f'PRSice2.GC.clumpr2.{c}.{covar_name[i]}'\n",
    "                PRSice2_new(gwas_path='../../sum_stats_data/gastric_cancer/hum0197.v3.BBJ.GC.v1/gwas.PRSice.txt',\n",
    "                                disease='gastric_cancer', clump_r2=c, output_prefix=prefix, covar_cols=covar_cols)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4. PD (Parkinson disease)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 17:50:28\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/pd/gwas.catalog.EAS.GCST90278092/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex,Age \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/pd/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/pd/PRSice2.PD.clumpr2.0.1.age \\\n",
      "    --pheno ../../results/pd/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/pd/gwas.catalog.EAS.GCST90278092/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tPOS\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "490637 variant(s) observed in base file, with: \n",
      "25069 ambiguous variant(s) excluded \n",
      "465568 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "5316425 variant(s) not found in previous data \n",
      "10 variant(s) with mismatch information \n",
      "288631 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/pd/samples.txt \n",
      "Column Name of Sample ID: 110839298924+110839298924 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 94621 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "Age\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 17:51:01\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/pd/gwas.catalog.EAS.GCST90278092/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/pd/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/pd/PRSice2.PD.clumpr2.0.1.no.age \\\n",
      "    --pheno ../../results/pd/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/pd/gwas.catalog.EAS.GCST90278092/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tPOS\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "490637 variant(s) observed in base file, with: \n",
      "25069 ambiguous variant(s) excluded \n",
      "465568 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "5316425 variant(s) not found in previous data \n",
      "10 variant(s) with mismatch information \n",
      "288631 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/pd/samples.txt \n",
      "Column Name of Sample ID: 110839298924+110839298924 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 94621 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 17:51:18\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/pd/gwas.catalog.EAS.GCST90278092/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.400000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex,Age \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/pd/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/pd/PRSice2.PD.clumpr2.0.4.age \\\n",
      "    --pheno ../../results/pd/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/pd/gwas.catalog.EAS.GCST90278092/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tPOS\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "490637 variant(s) observed in base file, with: \n",
      "25069 ambiguous variant(s) excluded \n",
      "465568 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "5316425 variant(s) not found in previous data \n",
      "10 variant(s) with mismatch information \n",
      "288631 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/pd/samples.txt \n",
      "Column Name of Sample ID: 110839298924+110839298924 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 171488 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "Age\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 17:51:52\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/pd/gwas.catalog.EAS.GCST90278092/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.400000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/pd/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/pd/PRSice2.PD.clumpr2.0.4.no.age \\\n",
      "    --pheno ../../results/pd/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/pd/gwas.catalog.EAS.GCST90278092/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tPOS\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "490637 variant(s) observed in base file, with: \n",
      "25069 ambiguous variant(s) excluded \n",
      "465568 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "5316425 variant(s) not found in previous data \n",
      "10 variant(s) with mismatch information \n",
      "288631 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/pd/samples.txt \n",
      "Column Name of Sample ID: 110839298924+110839298924 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 171488 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 17:52:11\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/pd/gwas.catalog.EAS.GCST90278092/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.700000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex,Age \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/pd/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/pd/PRSice2.PD.clumpr2.0.7.age \\\n",
      "    --pheno ../../results/pd/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/pd/gwas.catalog.EAS.GCST90278092/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tPOS\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "490637 variant(s) observed in base file, with: \n",
      "25069 ambiguous variant(s) excluded \n",
      "465568 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "5316425 variant(s) not found in previous data \n",
      "10 variant(s) with mismatch information \n",
      "288631 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/pd/samples.txt \n",
      "Column Name of Sample ID: 110839298924+110839298924 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 225049 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "Age\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 17:52:46\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/pd/gwas.catalog.EAS.GCST90278092/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.700000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/pd/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/pd/PRSice2.PD.clumpr2.0.7.no.age \\\n",
      "    --pheno ../../results/pd/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/pd/gwas.catalog.EAS.GCST90278092/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tPOS\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "490637 variant(s) observed in base file, with: \n",
      "25069 ambiguous variant(s) excluded \n",
      "465568 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "5316425 variant(s) not found in previous data \n",
      "10 variant(s) with mismatch information \n",
      "288631 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/pd/samples.txt \n",
      "Column Name of Sample ID: 110839298924+110839298924 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 225049 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 17:53:05\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/pd/gwas.catalog.EAS.GCST90278092/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.900000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex,Age \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/pd/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/pd/PRSice2.PD.clumpr2.0.9.age \\\n",
      "    --pheno ../../results/pd/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/pd/gwas.catalog.EAS.GCST90278092/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tPOS\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "490637 variant(s) observed in base file, with: \n",
      "25069 ambiguous variant(s) excluded \n",
      "465568 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "5316425 variant(s) not found in previous data \n",
      "10 variant(s) with mismatch information \n",
      "288631 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/pd/samples.txt \n",
      "Column Name of Sample ID: 110839298924+110839298924 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 249366 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "Age\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 17:53:41\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/pd/gwas.catalog.EAS.GCST90278092/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.900000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/pd/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/pd/PRSice2.PD.clumpr2.0.9.no.age \\\n",
      "    --pheno ../../results/pd/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/pd/gwas.catalog.EAS.GCST90278092/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tPOS\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "490637 variant(s) observed in base file, with: \n",
      "25069 ambiguous variant(s) excluded \n",
      "465568 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "5316425 variant(s) not found in previous data \n",
      "10 variant(s) with mismatch information \n",
      "288631 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/pd/samples.txt \n",
      "Column Name of Sample ID: 110839298924+110839298924 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 249366 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    }
   ],
   "source": [
    "# BBJ+\n",
    "covar_name = ['age', 'no.age']\n",
    "for c in clump_r2_list:\n",
    "        for i, covar_cols in enumerate(['@PC[1-10],Sex,Age', '@PC[1-10],Sex']):\n",
    "                prefix = f'PRSice2.PD.clumpr2.{c}.{covar_name[i]}'\n",
    "                PRSice2_new(gwas_path='../../sum_stats_data/pd/gwas.catalog.EAS.GCST90278092/gwas.PRSice.txt',\n",
    "                                disease='pd', clump_r2=c, output_prefix=prefix, covar_cols=covar_cols)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 5. CKD (Chronic Kidney Disease)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 17:54:00\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/ckd/hum0197.v3.BBJ.CRF.v1/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex,Age \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/ckd/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/ckd/PRSice2.CKD.clumpr2.0.1.age \\\n",
      "    --pheno ../../results/ckd/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/ckd/hum0197.v3.BBJ.CRF.v1/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "10372517 variant(s) observed in base file, with: \n",
      "1492766 ambiguous variant(s) excluded \n",
      "8879751 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "1430876 variant(s) not found in previous data \n",
      "4174190 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/ckd/samples.txt \n",
      "Column Name of Sample ID: 336799182800+336799182800 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 130477 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "Age\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 17:55:32\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/ckd/hum0197.v3.BBJ.CRF.v1/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/ckd/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/ckd/PRSice2.CKD.clumpr2.0.1.no.age \\\n",
      "    --pheno ../../results/ckd/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/ckd/hum0197.v3.BBJ.CRF.v1/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "10372517 variant(s) observed in base file, with: \n",
      "1492766 ambiguous variant(s) excluded \n",
      "8879751 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "1430876 variant(s) not found in previous data \n",
      "4174190 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/ckd/samples.txt \n",
      "Column Name of Sample ID: 336799182800+336799182800 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 130477 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 17:57:05\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/ckd/hum0197.v3.BBJ.CRF.v1/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.400000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex,Age \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/ckd/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/ckd/PRSice2.CKD.clumpr2.0.4.age \\\n",
      "    --pheno ../../results/ckd/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/ckd/hum0197.v3.BBJ.CRF.v1/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "10372517 variant(s) observed in base file, with: \n",
      "1492766 ambiguous variant(s) excluded \n",
      "8879751 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "1430876 variant(s) not found in previous data \n",
      "4174190 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/ckd/samples.txt \n",
      "Column Name of Sample ID: 336799182800+336799182800 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 293196 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "Age\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 17:59:03\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/ckd/hum0197.v3.BBJ.CRF.v1/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.400000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/ckd/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/ckd/PRSice2.CKD.clumpr2.0.4.no.age \\\n",
      "    --pheno ../../results/ckd/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/ckd/hum0197.v3.BBJ.CRF.v1/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "10372517 variant(s) observed in base file, with: \n",
      "1492766 ambiguous variant(s) excluded \n",
      "8879751 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "1430876 variant(s) not found in previous data \n",
      "4174190 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/ckd/samples.txt \n",
      "Column Name of Sample ID: 336799182800+336799182800 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 293196 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 18:00:58\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/ckd/hum0197.v3.BBJ.CRF.v1/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.700000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex,Age \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/ckd/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/ckd/PRSice2.CKD.clumpr2.0.7.age \\\n",
      "    --pheno ../../results/ckd/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/ckd/hum0197.v3.BBJ.CRF.v1/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "10372517 variant(s) observed in base file, with: \n",
      "1492766 ambiguous variant(s) excluded \n",
      "8879751 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "1430876 variant(s) not found in previous data \n",
      "4174190 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/ckd/samples.txt \n",
      "Column Name of Sample ID: 336799182800+336799182800 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 529360 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "Age\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 18:03:40\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/ckd/hum0197.v3.BBJ.CRF.v1/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.700000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/ckd/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/ckd/PRSice2.CKD.clumpr2.0.7.no.age \\\n",
      "    --pheno ../../results/ckd/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/ckd/hum0197.v3.BBJ.CRF.v1/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "10372517 variant(s) observed in base file, with: \n",
      "1492766 ambiguous variant(s) excluded \n",
      "8879751 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "1430876 variant(s) not found in previous data \n",
      "4174190 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/ckd/samples.txt \n",
      "Column Name of Sample ID: 336799182800+336799182800 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 529360 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 18:06:23\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/ckd/hum0197.v3.BBJ.CRF.v1/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.900000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex,Age \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/ckd/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/ckd/PRSice2.CKD.clumpr2.0.9.age \\\n",
      "    --pheno ../../results/ckd/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/ckd/hum0197.v3.BBJ.CRF.v1/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "10372517 variant(s) observed in base file, with: \n",
      "1492766 ambiguous variant(s) excluded \n",
      "8879751 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "1430876 variant(s) not found in previous data \n",
      "4174190 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/ckd/samples.txt \n",
      "Column Name of Sample ID: 336799182800+336799182800 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 882790 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "Age\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 18:10:15\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/ckd/hum0197.v3.BBJ.CRF.v1/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.900000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/ckd/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/ckd/PRSice2.CKD.clumpr2.0.9.no.age \\\n",
      "    --pheno ../../results/ckd/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/ckd/hum0197.v3.BBJ.CRF.v1/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "10372517 variant(s) observed in base file, with: \n",
      "1492766 ambiguous variant(s) excluded \n",
      "8879751 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "1430876 variant(s) not found in previous data \n",
      "4174190 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/ckd/samples.txt \n",
      "Column Name of Sample ID: 336799182800+336799182800 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 882790 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    }
   ],
   "source": [
    "# BBJ+\n",
    "covar_name = ['age', 'no.age']\n",
    "for c in clump_r2_list:\n",
    "        for i, covar_cols in enumerate(['@PC[1-10],Sex,Age', '@PC[1-10],Sex']):\n",
    "                prefix = f'PRSice2.CKD.clumpr2.{c}.{covar_name[i]}'\n",
    "                PRSice2_new(gwas_path='../../sum_stats_data/ckd/hum0197.v3.BBJ.CRF.v1/gwas.PRSice.txt',\n",
    "                                disease='ckd', clump_r2=c, output_prefix=prefix, covar_cols=covar_cols)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 6. Osteoporosis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 18:14:02\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/osteoporosis/hum0197.v3.BBJ.OP.v1/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex,Age \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/osteoporosis/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/osteoporosis/PRSice2.OP.clumpr2.0.1.age \\\n",
      "    --pheno ../../results/osteoporosis/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/osteoporosis/hum0197.v3.BBJ.OP.v1/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "10372510 variant(s) observed in base file, with: \n",
      "1492773 ambiguous variant(s) excluded \n",
      "8879737 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1059 founder(s) included \n",
      "\n",
      "1431055 variant(s) not found in previous data \n",
      "4174011 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/osteoporosis/samples.txt \n",
      "Column Name of Sample ID: 855561523000+855561523000 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 130507 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "51 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "Age\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1059 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 18:15:59\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/osteoporosis/hum0197.v3.BBJ.OP.v1/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/osteoporosis/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/osteoporosis/PRSice2.OP.clumpr2.0.1.no.age \\\n",
      "    --pheno ../../results/osteoporosis/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/osteoporosis/hum0197.v3.BBJ.OP.v1/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "10372510 variant(s) observed in base file, with: \n",
      "1492773 ambiguous variant(s) excluded \n",
      "8879737 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1059 founder(s) included \n",
      "\n",
      "1431055 variant(s) not found in previous data \n",
      "4174011 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/osteoporosis/samples.txt \n",
      "Column Name of Sample ID: 855561523000+855561523000 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 130507 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "51 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1059 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 18:17:28\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/osteoporosis/hum0197.v3.BBJ.OP.v1/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.400000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex,Age \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/osteoporosis/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/osteoporosis/PRSice2.OP.clumpr2.0.4.age \\\n",
      "    --pheno ../../results/osteoporosis/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/osteoporosis/hum0197.v3.BBJ.OP.v1/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "10372510 variant(s) observed in base file, with: \n",
      "1492773 ambiguous variant(s) excluded \n",
      "8879737 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1059 founder(s) included \n",
      "\n",
      "1431055 variant(s) not found in previous data \n",
      "4174011 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/osteoporosis/samples.txt \n",
      "Column Name of Sample ID: 855561523000+855561523000 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 293211 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "51 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "Age\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1059 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 18:19:47\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/osteoporosis/hum0197.v3.BBJ.OP.v1/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.400000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/osteoporosis/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/osteoporosis/PRSice2.OP.clumpr2.0.4.no.age \\\n",
      "    --pheno ../../results/osteoporosis/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/osteoporosis/hum0197.v3.BBJ.OP.v1/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "10372510 variant(s) observed in base file, with: \n",
      "1492773 ambiguous variant(s) excluded \n",
      "8879737 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1059 founder(s) included \n",
      "\n",
      "1431055 variant(s) not found in previous data \n",
      "4174011 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/osteoporosis/samples.txt \n",
      "Column Name of Sample ID: 855561523000+855561523000 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 293211 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "51 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1059 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 18:21:41\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/osteoporosis/hum0197.v3.BBJ.OP.v1/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.700000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex,Age \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/osteoporosis/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/osteoporosis/PRSice2.OP.clumpr2.0.7.age \\\n",
      "    --pheno ../../results/osteoporosis/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/osteoporosis/hum0197.v3.BBJ.OP.v1/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "10372510 variant(s) observed in base file, with: \n",
      "1492773 ambiguous variant(s) excluded \n",
      "8879737 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1059 founder(s) included \n",
      "\n",
      "1431055 variant(s) not found in previous data \n",
      "4174011 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/osteoporosis/samples.txt \n",
      "Column Name of Sample ID: 855561523000+855561523000 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 529309 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "51 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "Age\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1059 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 18:24:38\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/osteoporosis/hum0197.v3.BBJ.OP.v1/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.700000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/osteoporosis/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/osteoporosis/PRSice2.OP.clumpr2.0.7.no.age \\\n",
      "    --pheno ../../results/osteoporosis/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/osteoporosis/hum0197.v3.BBJ.OP.v1/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "10372510 variant(s) observed in base file, with: \n",
      "1492773 ambiguous variant(s) excluded \n",
      "8879737 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1059 founder(s) included \n",
      "\n",
      "1431055 variant(s) not found in previous data \n",
      "4174011 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/osteoporosis/samples.txt \n",
      "Column Name of Sample ID: 855561523000+855561523000 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 529309 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "51 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1059 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 18:27:10\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/osteoporosis/hum0197.v3.BBJ.OP.v1/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.900000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex,Age \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/osteoporosis/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/osteoporosis/PRSice2.OP.clumpr2.0.9.age \\\n",
      "    --pheno ../../results/osteoporosis/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/osteoporosis/hum0197.v3.BBJ.OP.v1/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "10372510 variant(s) observed in base file, with: \n",
      "1492773 ambiguous variant(s) excluded \n",
      "8879737 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1059 founder(s) included \n",
      "\n",
      "1431055 variant(s) not found in previous data \n",
      "4174011 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/osteoporosis/samples.txt \n",
      "Column Name of Sample ID: 855561523000+855561523000 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 883097 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "51 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "Age\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1059 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 18:31:13\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/osteoporosis/hum0197.v3.BBJ.OP.v1/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.900000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/osteoporosis/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/osteoporosis/PRSice2.OP.clumpr2.0.9.no.age \\\n",
      "    --pheno ../../results/osteoporosis/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/osteoporosis/hum0197.v3.BBJ.OP.v1/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "10372510 variant(s) observed in base file, with: \n",
      "1492773 ambiguous variant(s) excluded \n",
      "8879737 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1059 founder(s) included \n",
      "\n",
      "1431055 variant(s) not found in previous data \n",
      "4174011 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/osteoporosis/samples.txt \n",
      "Column Name of Sample ID: 855561523000+855561523000 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 883097 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "51 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1059 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    }
   ],
   "source": [
    "# BBJ+\n",
    "covar_name = ['age', 'no.age']\n",
    "for c in clump_r2_list:\n",
    "        for i, covar_cols in enumerate(['@PC[1-10],Sex,Age', '@PC[1-10],Sex']):\n",
    "                prefix = f'PRSice2.OP.clumpr2.{c}.{covar_name[i]}'\n",
    "                PRSice2_new(gwas_path='../../sum_stats_data/osteoporosis/hum0197.v3.BBJ.OP.v1/gwas.PRSice.txt',\n",
    "                                disease='osteoporosis', clump_r2=c, output_prefix=prefix, covar_cols=covar_cols)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 7. CAD (Coronary Artery Disease)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 18:34:49\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/cad/gwas.BBJ.CAD/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex,Age \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/cad/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/cad/PRSice2.CAD.clumpr2.0.1.age \\\n",
      "    --pheno ../../results/cad/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/cad/gwas.BBJ.CAD/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tPOS\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "5074555 variant(s) observed in base file, with: \n",
      "778532 ambiguous variant(s) excluded \n",
      "4296023 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "1471791 variant(s) not found in previous data \n",
      "84 variant(s) with mismatch information \n",
      "4133191 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/cad/samples.txt \n",
      "Column Name of Sample ID: 094510528500+094510528500 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 102628 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "Age\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 18:36:18\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/cad/gwas.BBJ.CAD/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/cad/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/cad/PRSice2.CAD.clumpr2.0.1.no.age \\\n",
      "    --pheno ../../results/cad/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/cad/gwas.BBJ.CAD/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tPOS\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "5074555 variant(s) observed in base file, with: \n",
      "778532 ambiguous variant(s) excluded \n",
      "4296023 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "1471791 variant(s) not found in previous data \n",
      "84 variant(s) with mismatch information \n",
      "4133191 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/cad/samples.txt \n",
      "Column Name of Sample ID: 094510528500+094510528500 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 102628 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 18:37:21\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/cad/gwas.BBJ.CAD/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.400000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex,Age \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/cad/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/cad/PRSice2.CAD.clumpr2.0.4.age \\\n",
      "    --pheno ../../results/cad/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/cad/gwas.BBJ.CAD/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tPOS\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "5074555 variant(s) observed in base file, with: \n",
      "778532 ambiguous variant(s) excluded \n",
      "4296023 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "1471791 variant(s) not found in previous data \n",
      "84 variant(s) with mismatch information \n",
      "4133191 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/cad/samples.txt \n",
      "Column Name of Sample ID: 094510528500+094510528500 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 243284 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "Age\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 18:39:11\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/cad/gwas.BBJ.CAD/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.400000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/cad/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/cad/PRSice2.CAD.clumpr2.0.4.no.age \\\n",
      "    --pheno ../../results/cad/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/cad/gwas.BBJ.CAD/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tPOS\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "5074555 variant(s) observed in base file, with: \n",
      "778532 ambiguous variant(s) excluded \n",
      "4296023 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "1471791 variant(s) not found in previous data \n",
      "84 variant(s) with mismatch information \n",
      "4133191 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/cad/samples.txt \n",
      "Column Name of Sample ID: 094510528500+094510528500 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 243284 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 18:40:33\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/cad/gwas.BBJ.CAD/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.700000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex,Age \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/cad/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/cad/PRSice2.CAD.clumpr2.0.7.age \\\n",
      "    --pheno ../../results/cad/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/cad/gwas.BBJ.CAD/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tPOS\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "5074555 variant(s) observed in base file, with: \n",
      "778532 ambiguous variant(s) excluded \n",
      "4296023 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "1471791 variant(s) not found in previous data \n",
      "84 variant(s) with mismatch information \n",
      "4133191 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/cad/samples.txt \n",
      "Column Name of Sample ID: 094510528500+094510528500 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 461620 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "Age\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 18:43:01\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/cad/gwas.BBJ.CAD/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.700000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/cad/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/cad/PRSice2.CAD.clumpr2.0.7.no.age \\\n",
      "    --pheno ../../results/cad/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/cad/gwas.BBJ.CAD/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tPOS\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "5074555 variant(s) observed in base file, with: \n",
      "778532 ambiguous variant(s) excluded \n",
      "4296023 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "1471791 variant(s) not found in previous data \n",
      "84 variant(s) with mismatch information \n",
      "4133191 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/cad/samples.txt \n",
      "Column Name of Sample ID: 094510528500+094510528500 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 461620 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 18:44:59\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/cad/gwas.BBJ.CAD/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.900000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex,Age \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/cad/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/cad/PRSice2.CAD.clumpr2.0.9.age \\\n",
      "    --pheno ../../results/cad/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/cad/gwas.BBJ.CAD/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tPOS\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "5074555 variant(s) observed in base file, with: \n",
      "778532 ambiguous variant(s) excluded \n",
      "4296023 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "1471791 variant(s) not found in previous data \n",
      "84 variant(s) with mismatch information \n",
      "4133191 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/cad/samples.txt \n",
      "Column Name of Sample ID: 094510528500+094510528500 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 802134 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "Age\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 18:48:23\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/cad/gwas.BBJ.CAD/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.900000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/cad/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/cad/PRSice2.CAD.clumpr2.0.9.no.age \\\n",
      "    --pheno ../../results/cad/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/cad/gwas.BBJ.CAD/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tPOS\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "5074555 variant(s) observed in base file, with: \n",
      "778532 ambiguous variant(s) excluded \n",
      "4296023 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "1471791 variant(s) not found in previous data \n",
      "84 variant(s) with mismatch information \n",
      "4133191 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/cad/samples.txt \n",
      "Column Name of Sample ID: 094510528500+094510528500 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 802134 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    }
   ],
   "source": [
    "# BBJ+\n",
    "covar_name = ['age', 'no.age']\n",
    "for c in clump_r2_list:\n",
    "        for i, covar_cols in enumerate(['@PC[1-10],Sex,Age', '@PC[1-10],Sex']):\n",
    "                prefix = f'PRSice2.CAD.clumpr2.{c}.{covar_name[i]}'\n",
    "                PRSice2_new(gwas_path='../../sum_stats_data/cad/gwas.BBJ.CAD/gwas.PRSice.txt',\n",
    "                                disease='cad', clump_r2=c, output_prefix=prefix, covar_cols=covar_cols)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 8. Hyperlipidemia"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 18:51:19\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/hyperlipidemia/gwas.catalog.EAS.GCST90090994/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex,Age \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/hyperlipidemia/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/hyperlipidemia/PRSice2.HL.clumpr2.0.1.age \\\n",
      "    --pheno ../../results/hyperlipidemia/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/hyperlipidemia/gwas.catalog.EAS.GCST90090994/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "592691 variant(s) observed in base file, with: \n",
      "37843 ambiguous variant(s) excluded \n",
      "554848 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "5145347 variant(s) not found in previous data \n",
      "10 variant(s) with mismatch information \n",
      "459709 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/hyperlipidemia/samples.txt \n",
      "Column Name of Sample ID: 711597776200+711597776200 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 75063 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "Age\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 18:51:37\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/hyperlipidemia/gwas.catalog.EAS.GCST90090994/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/hyperlipidemia/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/hyperlipidemia/PRSice2.HL.clumpr2.0.1.no.age \\\n",
      "    --pheno ../../results/hyperlipidemia/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/hyperlipidemia/gwas.catalog.EAS.GCST90090994/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "592691 variant(s) observed in base file, with: \n",
      "37843 ambiguous variant(s) excluded \n",
      "554848 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "5145347 variant(s) not found in previous data \n",
      "10 variant(s) with mismatch information \n",
      "459709 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/hyperlipidemia/samples.txt \n",
      "Column Name of Sample ID: 711597776200+711597776200 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 75063 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 18:51:55\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/hyperlipidemia/gwas.catalog.EAS.GCST90090994/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.400000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex,Age \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/hyperlipidemia/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/hyperlipidemia/PRSice2.HL.clumpr2.0.4.age \\\n",
      "    --pheno ../../results/hyperlipidemia/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/hyperlipidemia/gwas.catalog.EAS.GCST90090994/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "592691 variant(s) observed in base file, with: \n",
      "37843 ambiguous variant(s) excluded \n",
      "554848 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "5145347 variant(s) not found in previous data \n",
      "10 variant(s) with mismatch information \n",
      "459709 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/hyperlipidemia/samples.txt \n",
      "Column Name of Sample ID: 711597776200+711597776200 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 165593 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "Age\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 18:52:15\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/hyperlipidemia/gwas.catalog.EAS.GCST90090994/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.400000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/hyperlipidemia/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/hyperlipidemia/PRSice2.HL.clumpr2.0.4.no.age \\\n",
      "    --pheno ../../results/hyperlipidemia/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/hyperlipidemia/gwas.catalog.EAS.GCST90090994/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "592691 variant(s) observed in base file, with: \n",
      "37843 ambiguous variant(s) excluded \n",
      "554848 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "5145347 variant(s) not found in previous data \n",
      "10 variant(s) with mismatch information \n",
      "459709 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/hyperlipidemia/samples.txt \n",
      "Column Name of Sample ID: 711597776200+711597776200 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 165593 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 18:52:35\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/hyperlipidemia/gwas.catalog.EAS.GCST90090994/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.700000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex,Age \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/hyperlipidemia/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/hyperlipidemia/PRSice2.HL.clumpr2.0.7.age \\\n",
      "    --pheno ../../results/hyperlipidemia/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/hyperlipidemia/gwas.catalog.EAS.GCST90090994/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "592691 variant(s) observed in base file, with: \n",
      "37843 ambiguous variant(s) excluded \n",
      "554848 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "5145347 variant(s) not found in previous data \n",
      "10 variant(s) with mismatch information \n",
      "459709 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/hyperlipidemia/samples.txt \n",
      "Column Name of Sample ID: 711597776200+711597776200 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 270782 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "Age\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 18:52:58\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/hyperlipidemia/gwas.catalog.EAS.GCST90090994/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.700000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/hyperlipidemia/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/hyperlipidemia/PRSice2.HL.clumpr2.0.7.no.age \\\n",
      "    --pheno ../../results/hyperlipidemia/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/hyperlipidemia/gwas.catalog.EAS.GCST90090994/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "592691 variant(s) observed in base file, with: \n",
      "37843 ambiguous variant(s) excluded \n",
      "554848 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "5145347 variant(s) not found in previous data \n",
      "10 variant(s) with mismatch information \n",
      "459709 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/hyperlipidemia/samples.txt \n",
      "Column Name of Sample ID: 711597776200+711597776200 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 270782 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 18:53:20\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/hyperlipidemia/gwas.catalog.EAS.GCST90090994/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.900000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex,Age \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/hyperlipidemia/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/hyperlipidemia/PRSice2.HL.clumpr2.0.9.age \\\n",
      "    --pheno ../../results/hyperlipidemia/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/hyperlipidemia/gwas.catalog.EAS.GCST90090994/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "592691 variant(s) observed in base file, with: \n",
      "37843 ambiguous variant(s) excluded \n",
      "554848 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "5145347 variant(s) not found in previous data \n",
      "10 variant(s) with mismatch information \n",
      "459709 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/hyperlipidemia/samples.txt \n",
      "Column Name of Sample ID: 711597776200+711597776200 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 352377 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "Age\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 18:53:44\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/hyperlipidemia/gwas.catalog.EAS.GCST90090994/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.900000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/hyperlipidemia/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/hyperlipidemia/PRSice2.HL.clumpr2.0.9.no.age \\\n",
      "    --pheno ../../results/hyperlipidemia/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/hyperlipidemia/gwas.catalog.EAS.GCST90090994/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "592691 variant(s) observed in base file, with: \n",
      "37843 ambiguous variant(s) excluded \n",
      "554848 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "5145347 variant(s) not found in previous data \n",
      "10 variant(s) with mismatch information \n",
      "459709 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/hyperlipidemia/samples.txt \n",
      "Column Name of Sample ID: 711597776200+711597776200 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 352377 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    }
   ],
   "source": [
    "# BBJ+\n",
    "covar_name = ['age', 'no.age']\n",
    "for c in clump_r2_list:\n",
    "        for i, covar_cols in enumerate(['@PC[1-10],Sex,Age', '@PC[1-10],Sex']):\n",
    "                prefix = f'PRSice2.HL.clumpr2.{c}.{covar_name[i]}'\n",
    "                PRSice2_new(gwas_path='../../sum_stats_data/hyperlipidemia/gwas.catalog.EAS.GCST90090994/gwas.PRSice.txt',\n",
    "                                disease='hyperlipidemia', clump_r2=c, output_prefix=prefix, covar_cols=covar_cols)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 9. Osteoarthritis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 18:54:08\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/osteoarthritis/gwas.catalog.EAS.GCST90134281/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex,Age \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/osteoarthritis/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/osteoarthritis/PRSice2.OA.clumpr2.0.1.age \\\n",
      "    --pheno ../../results/osteoarthritis/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/osteoarthritis/gwas.catalog.EAS.GCST90134281/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tPOS\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "5261232 variant(s) observed in base file, with: \n",
      "808049 ambiguous variant(s) excluded \n",
      "4453183 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "1328432 variant(s) not found in previous data \n",
      "118 variant(s) with mismatch information \n",
      "4276516 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/osteoarthritis/samples.txt \n",
      "Column Name of Sample ID: 853435767400+853435767400 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 115079 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "Age\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 18:55:15\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/osteoarthritis/gwas.catalog.EAS.GCST90134281/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/osteoarthritis/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/osteoarthritis/PRSice2.OA.clumpr2.0.1.no.age \\\n",
      "    --pheno ../../results/osteoarthritis/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/osteoarthritis/gwas.catalog.EAS.GCST90134281/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tPOS\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "5261232 variant(s) observed in base file, with: \n",
      "808049 ambiguous variant(s) excluded \n",
      "4453183 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "1328432 variant(s) not found in previous data \n",
      "118 variant(s) with mismatch information \n",
      "4276516 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/osteoarthritis/samples.txt \n",
      "Column Name of Sample ID: 853435767400+853435767400 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 115079 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 18:56:19\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/osteoarthritis/gwas.catalog.EAS.GCST90134281/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.400000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex,Age \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/osteoarthritis/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/osteoarthritis/PRSice2.OA.clumpr2.0.4.age \\\n",
      "    --pheno ../../results/osteoarthritis/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/osteoarthritis/gwas.catalog.EAS.GCST90134281/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tPOS\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "5261232 variant(s) observed in base file, with: \n",
      "808049 ambiguous variant(s) excluded \n",
      "4453183 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "1328432 variant(s) not found in previous data \n",
      "118 variant(s) with mismatch information \n",
      "4276516 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/osteoarthritis/samples.txt \n",
      "Column Name of Sample ID: 853435767400+853435767400 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 267111 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "Age\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 18:57:48\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/osteoarthritis/gwas.catalog.EAS.GCST90134281/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.400000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/osteoarthritis/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/osteoarthritis/PRSice2.OA.clumpr2.0.4.no.age \\\n",
      "    --pheno ../../results/osteoarthritis/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/osteoarthritis/gwas.catalog.EAS.GCST90134281/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tPOS\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "5261232 variant(s) observed in base file, with: \n",
      "808049 ambiguous variant(s) excluded \n",
      "4453183 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "1328432 variant(s) not found in previous data \n",
      "118 variant(s) with mismatch information \n",
      "4276516 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/osteoarthritis/samples.txt \n",
      "Column Name of Sample ID: 853435767400+853435767400 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 267111 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 18:59:14\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/osteoarthritis/gwas.catalog.EAS.GCST90134281/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.700000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex,Age \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/osteoarthritis/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/osteoarthritis/PRSice2.OA.clumpr2.0.7.age \\\n",
      "    --pheno ../../results/osteoarthritis/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/osteoarthritis/gwas.catalog.EAS.GCST90134281/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tPOS\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "5261232 variant(s) observed in base file, with: \n",
      "808049 ambiguous variant(s) excluded \n",
      "4453183 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "1328432 variant(s) not found in previous data \n",
      "118 variant(s) with mismatch information \n",
      "4276516 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/osteoarthritis/samples.txt \n",
      "Column Name of Sample ID: 853435767400+853435767400 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 496204 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "Age\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 19:01:21\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/osteoarthritis/gwas.catalog.EAS.GCST90134281/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.700000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/osteoarthritis/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/osteoarthritis/PRSice2.OA.clumpr2.0.7.no.age \\\n",
      "    --pheno ../../results/osteoarthritis/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/osteoarthritis/gwas.catalog.EAS.GCST90134281/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tPOS\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "5261232 variant(s) observed in base file, with: \n",
      "808049 ambiguous variant(s) excluded \n",
      "4453183 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "1328432 variant(s) not found in previous data \n",
      "118 variant(s) with mismatch information \n",
      "4276516 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/osteoarthritis/samples.txt \n",
      "Column Name of Sample ID: 853435767400+853435767400 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 496204 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 19:03:26\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/osteoarthritis/gwas.catalog.EAS.GCST90134281/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.900000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex,Age \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/osteoarthritis/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/osteoarthritis/PRSice2.OA.clumpr2.0.9.age \\\n",
      "    --pheno ../../results/osteoarthritis/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/osteoarthritis/gwas.catalog.EAS.GCST90134281/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tPOS\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "5261232 variant(s) observed in base file, with: \n",
      "808049 ambiguous variant(s) excluded \n",
      "4453183 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "1328432 variant(s) not found in previous data \n",
      "118 variant(s) with mismatch information \n",
      "4276516 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/osteoarthritis/samples.txt \n",
      "Column Name of Sample ID: 853435767400+853435767400 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 851079 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "Age\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2024-12-17 19:06:37\n",
      "./PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../../sum_stats_data/osteoarthritis/gwas.catalog.EAS.GCST90134281/gwas.PRSice.txt \\\n",
      "    --beta  \\\n",
      "    --binary-target T \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.900000 \\\n",
      "    --cov ../../target_data/covar_list.tsv \\\n",
      "    --cov-col @PC[1-10],Sex \\\n",
      "    --interval 5e-05 \\\n",
      "    --keep ../../results/osteoarthritis/samples.txt \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ../../results/osteoarthritis/PRSice2.OA.clumpr2.0.9.no.age \\\n",
      "    --pheno ../../results/osteoarthritis/samples.txt \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --score sum \\\n",
      "    --seed 68 \\\n",
      "    --snp SNP \\\n",
      "    --stat BETA \\\n",
      "    --target ../../target_data/target.data.impute \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: \n",
      "../../target_data/target.data.impute (bed) \n",
      "\n",
      "Start processing gwas.PRSice \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "../../sum_stats_data/osteoarthritis/gwas.catalog.EAS.GCST90134281/gwas.PRSice.txt \n",
      "Header of file is: \n",
      "CHR\tSNP\tPOS\tA1\tA2\tBETA\tP \n",
      "\n",
      "Reading 100.00%\n",
      "5261232 variant(s) observed in base file, with: \n",
      "808049 ambiguous variant(s) excluded \n",
      "4453183 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1608 people (0 male(s), 0 female(s)) observed \n",
      "1058 founder(s) included \n",
      "\n",
      "1328432 variant(s) not found in previous data \n",
      "118 variant(s) with mismatch information \n",
      "4276516 variant(s) included \n",
      "\n",
      "Phenotype file: ../../results/osteoarthritis/samples.txt \n",
      "Column Name of Sample ID: 853435767400+853435767400 \n",
      "Note: If the phenotype file does not contain a header, the \n",
      "column name will be displayed as the Sample ID which is \n",
      "expected. \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 851079 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1008 control(s) \n",
      "50 case(s) \n",
      "\n",
      "Processing the covariate file: \n",
      "../../target_data/covar_list.tsv \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "Sex\t0\t- \n",
      "PC1\t0\t- \n",
      "PC2\t0\t- \n",
      "PC3\t0\t- \n",
      "PC4\t0\t- \n",
      "PC5\t0\t- \n",
      "PC6\t0\t- \n",
      "PC7\t0\t- \n",
      "PC8\t0\t- \n",
      "PC9\t0\t- \n",
      "PC10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1058 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n"
     ]
    }
   ],
   "source": [
    "# BBJ+\n",
    "covar_name = ['age', 'no.age']\n",
    "for c in clump_r2_list:\n",
    "        for i, covar_cols in enumerate(['@PC[1-10],Sex,Age', '@PC[1-10],Sex']):\n",
    "                prefix = f'PRSice2.OA.clumpr2.{c}.{covar_name[i]}'\n",
    "                PRSice2_new(gwas_path='../../sum_stats_data/osteoarthritis/gwas.catalog.EAS.GCST90134281/gwas.PRSice.txt',\n",
    "                                disease='osteoarthritis', clump_r2=c, output_prefix=prefix, covar_cols=covar_cols)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "vgp",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
